main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -450,6 +450,9 @@ object Safety {
       )
 
       (Type.eraseAliases(from).baseType, to.map(Type.eraseAliases).map(_.baseType)) match {
+        // Allow all null casts
+        case (Type.Null, _) => ()
```

Comment:
This is clearly a bit weird, but this PR is a pure refactor - we can fix this in follow up work

Reply:
What is the fix?

Reply:
Just that e.g. `unchecked_cast(null as Int32)` should probably also give a safety error